### PR TITLE
Added Plot Options to Box Plot dialog

### LINKED ISSUE #7556: Change the specific plots to the new system
I am suggesting @EstherNjeriLiberatta for this, as (I am afraid) you did the PICSA and mapping graphs very well!

a) I suggest there be a single button, just `Plot Options`.  And it has the same pull-down as for the mapping dialogue.  
b) Then the first option is also Plot Options.
c) This is followed by each of the remaining options for that dialogue, and they could be in alphabetical order.
d) I suggest they don't change for the different options at the top.  So, for example, for bar charts it is the same whether the top radio button if Frequency, Value, Treemap or Wordcloud.
e) Then simply include each of those, including label and lollipop, etc.

### PR DESCRIPTION
**Pull request by derekagorhom**
partly fixes #7556 
Replaces #7570 

This PR is ready for review.

### CODE REVIEW

File: instat/dlgBoxPlot.vb [hunk 1]
```diff
@@ -317,122 +325,149 @@ Public Class dlgBoxplot
...
+    Private Sub ucrChkGrouptoConnect_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkGrouptoConnect.ControlValueChanged
+        If ucrChkGrouptoConnect.Checked Then
+            clsBaseOperator.AddParameter(strStatSummaryParameterName, clsRFunctionParameter:=clsStatSummary, iPosition:=4)
         Else
-            cmdBoxPlotOptions.Enabled = False
-            cmdOptions.Enabled = False
+            clsBaseOperator.RemoveParameterByName(strStatSummaryParameterName)
         End If
     End Sub
+    Private Sub openSdgLayerOptions(clsNewGeomFunc As RFunction)
```
Comment (lloyddewit) [inline instat/dlgBoxPlot.vb:371]:
```suggestion

    Private Sub openSdgLayerOptions(clsNewGeomFunc As RFunction)
```

File: instat/dlgBoxPlot.vb [hunk 2]
```diff
@@ -317,122 +325,149 @@ Public Class dlgBoxplot
...
+
+    Private Sub DialogueSize()
+        If rdoBoxplotTufte.Checked Then
+            Me.Size = New Size(441, 505)
+            Me.ucrSaveBoxplot.Location = New Point(10, 384)
+            Me.ucrBase.Location = New Point(10, 413)
+        ElseIf rdoViolin.Checked Then
+            Me.Size = New Size(441, 505)
+            Me.ucrSaveBoxplot.Location = New Point(10, 384)
+            Me.ucrBase.Location = New Point(10, 413)
```
Comment (lloyddewit) [inline instat/dlgBoxPlot.vb:450]: These statements look the same as lines 444 to 446.
Could we remove the `ElseIf` and replace with an `OrElse` on line 443?

File: instat/dlgBoxPlot.vb [hunk 3]
```diff
@@ -317,122 +325,149 @@ Public Class dlgBoxplot
...
 
+    Private Sub ucrSaveBoxplot_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrSaveBoxplot.ControlContentsChanged, ucrVariablesAsFactorForBoxplot.ControlContentsChanged
+        TestOkEnabled()
+    End Sub
+
+    Private Sub EnableDisableBoxplotOptions()
+        toolStripMenuItemJitterOptions.Enabled = rdoJitter.Checked OrElse ((rdoBoxplotTufte.Checked _
+            OrElse rdoViolin.Checked) AndAlso ucrChkAddPoints.Checked)
+        toolStripMenuItemViolinOptions.Enabled = rdoViolin.Checked
+        ToolStripMenuItemSummaryOptions.Enabled = rdoBoxplotTufte.Checked OrElse ((rdoJitter.Checked _
```
Comment (lloyddewit) [inline instat/dlgBoxPlot.vb:466]: rename to `toolStripMenuItemSummaryOptions` to make consistent?

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@derekagorhom looks good, just a couple of questions

**Comment by derekagorhom** [comment]
thanks @lloyddewit, i have made the changes

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@berylwaswa if you can test, then we can merge, thanks

**Review by berylwaswa — APPROVED** [review]
@derekagorhom @lloyddewit the options are all working fine.

**Review by rdstern — APPROVED** [review]
Looks good

**Pull request merged**